### ISSUE TITLE
Confusing UX with the tutorial and screen readers

### ISSUE THREAD

**Issue by srallen** [zooniverse/Panoptes-Front-End#5531]
## Current behavior
_Please include any error messages from the browser console and/or screenshots_

Original report:
https://www.zooniverse.org/projects/anaelisa24/sounds-of-new-york-city-sonyc/talk/1795/1143269?comment=1867713

https://www.zooniverse.org/talk/18/1145904

> Hi, I'm totally blind and thought I'd try your project. I'm using Windows 10 and the NVDA screen reader. First I tried Firefox, that was a disaster, it froze my screen reader when I pressed the done and talk button, so I had to bring up Narrator and close Firefox completely. Also many things were hidden. I then tried Chrome and discovered I was missing important things with Firefox like the near and far questions entirely. They just weren't showing. So with Chrome I can play a sound, read through the choices, and pick one and click "Identify" and go on to the next sound I heard in the recording. After I've identified all the sounds I can press done and listen to another sound. But there's one problem. There's something on the page that seems to do absolutely nothing, and that is the 7 radio buttons for steps 1 through 7. I can't even guess what these are for because when I select #2 nothing changes. Also there is a button labelled task, and one labelled tutorial. These buttons also seem to do nothing. I thought when I pressed the tutorial button I would find a tutorial about how the site works, and if there's anything I need to do that I am missing, but I found no such thing. The one time I pressed done and talk and went to the talk page I thought that was what you were supposed to do each time you were done with a sound. One more thing. When using Chrome I identified a reverse beep, and it said something about the sound graph and how a reverse beep should look, and of course I can't see that, but it was obviously a truck backing up that I heard. But the curious thing was that there were 2 radio buttons for the reverse beep with no explanation or alt-text explaining what those were supposed to be. They just said 1 and 2. I assume there are supposed to be 2 kinds of reverse beeps and I'm supposed to identify which type it is, but I had no idea what to do there so I just left the 1 radio button checked.
Thanks.

## Additional information

- **Operating system:**
Windows 10
- **Browser:**
Firefox and Chrome with NVDA (https://www.nvaccess.org/)

**Comment by eatyourgreens** [comment]
There’s a few different things going on here. 
- NVDA + Firefox misses a bunch of stuff, including the near and far questions for each sound. 
- Task and Tutorial aren’t marked up as tabs.

**Comment by eatyourgreens** [comment]
The tutorial dialog doesn't have `role="dialog"`, which seems to be a bug in modal-form as the role is specified here: https://github.com/zooniverse/modal-form/blob/b6e0e8e527f4c578cd42cc6961a0c18b5f776b61/dialog.js#L89-L96

**Comment by srallen** [comment]
We could move the tutorial from the new classifier to the react-components package and publish it. Then switch to that one here to the new one? This would also necessitate refactoring the http requests into a redux store here too I think.

**Comment by eatyourgreens** [comment]
I think the new one still needs the fixes that are implemented here, but that sounds good.

Would that also mean getting the tutorial tab to work in PFE? I notice that the Task button does nothing in PFE (and the Tutorial button appears to do nothing) because PFE doesn't have the tutorial in a tab panel.

**Comment by srallen** [comment]
Ah, I didn't realize the new tutorial has the same issue. Swapping the tutorial was what I saw as the other option if it was easier to fix it there and/or it was already working as expected.

**Comment by eatyourgreens** [comment]
#5535 is just some small markup changes to make the radio button labels more explicit, and to try and force the browser to announce when the tutorial content changes. Plus a bunch of linting.

Working on it, I realised we've essentially built a tabbed component which is presented as a tab panel inside another tabbed component.

**Comment by eatyourgreens** [comment]
I just came across this, from the same volunteer.

>The step 1 through 4 radio buttons are always on the page whether I click tutorial or not, but when I click tutorial they seem to duplicate themselves down the page at least 3 times.

https://www.zooniverse.org/projects/vivitang/earthquake-detective/talk/1903/1140957?comment=1870674&page=1

I wonder if we're rendering duplicate tutorial controls for each step? Interesting that stuff is hidden visually but not hidden from the DOM.

**Comment by eatyourgreens** [comment]
I've opened #5541 for the specific problem of navigating the tutorial. From what I remember, the new tutorial uses exactly the same setup: radio buttons beneath the tutorial to select the current step.

**Comment by eatyourgreens** [comment]
@srallen if you have a chance to look at the changes to the tutorial in #5535 that would be great. I think the PFE tutorial now behaves more like the new tutorial and some of the changes I've made there will need to be copied across to the new classifier.

**Comment by eatyourgreens** [comment]
Wrapping the tutorial in `<div role='dialog'>` reads out the content when the dialog opens. That role is already declared in the modal-form code, but is never added to the rendered HTML. I'm not going to fix that modal-form bug here, as I believe it's already fixed in the new tutorial.

**Comment by eatyourgreens** [comment]
Working on the tutorial again today, I realised that the mini-course uses almost exactly the same code so will have similar access issues.

**Comment by srallen** [comment]
Sorry I'm not gonna have time to look at this today.
